packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,255 @@
+   *
+   * @default Duration.seconds(0)
+   */
+  readonly timeToLive?: cdk.Duration;
+
+  /**
+   * How often the canary will run during its lifetime. The syntax for expression is 'rate(number unit)'
+   * where unit can be 'minute', 'minutes', or 'hour'. You can specify a frequency between 'rate(1 minute)'
+   * and 'rate(1 hour)'.
+   *
+   * The special expression 'rate(0 minute)' specifies that the canary will run only once when it is started.
+   *
+   * @default 'rate(5 minutes)'
+   */
+  readonly frequency?: cdk.Duration;
+
+  /**
+   * Whether or not the canary should start after creation.
+   *
+   * @default true
+   */
+  readonly enableCanary?: boolean;
+
+  /**
+   * How many days should successful runs be retained
+   *
+   * @default Duration.days(31)
+   */
+  readonly successRetentionPeriod?: cdk.Duration;
+
+  /**
+   * How many days should failed runs be retained
+   *
+   * @default Duration.days(31)
+   */
+  readonly failureRetentionPeriod?: cdk.Duration;
+
+}
+
+/**
+ * Properties of a canary
+ */
+export interface CanaryProps extends CanaryOptions {
+
+  /**
+   * Specify the endpoint that you want the canary code to hit. Alternatively, you can specify
+   * your own canary script to run.
+   *
+   * This is not yet implemented.
+   */
+  //readonly test: Test;
+
+  /**
+   * The name of the canary. This constitutes the physical ID of the canary.
+   *
+   * @default - A unique physical ID will be generated for you and used as the canary name.
+   */
+  readonly name: string;
+}
+
+export class Canary extends CanaryBase {
+  /**
+   * Execution role associated with this Canary.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Canary ID
+   * @attribute
+   */
+  public readonly canaryId: string;
+
+  /**
+   * Canary State
+   * @attribute
+   */
+  public readonly canaryState: string;
+
+  /**
+   * Canary Name
+   * @attribute
+   */
+  public readonly canaryName: string;
+
+  constructor(scope: cdk.Construct, id: string, props: CanaryProps) {
+    super(scope, id);
+
+    const s3Location = props.artifactBucket?.s3UrlForObject() ?? new s3.Bucket(this, 'ServiceBucket').s3UrlForObject();
+
+    // Created role will need these policies to run the Canary.
+    const policy = new iam.PolicyDocument({
+      statements: [new iam.PolicyStatement({
+        resources: ['*'],
+        actions: [
+          's3:PutObject',
+          's3:GetBucketLocation',
+          's3:ListAllMyBuckets',
+          'cloudwatch:PutMetricData',
+          'logs:CreateLogGroup',
+          'logs:CreateLogStream',
+          'logs:PutLogEvents',
+        ],
+      })],
+    });
+    const inlinePolicies = { canaryPolicy: policy };
+
+    this.role = props.role ?? new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+      inlinePolicies,
+    });
+
+    const duration = props.timeToLive ?? cdk.Duration.seconds(0);
+    const expression = props.frequency ?? cdk.Duration.minutes(5);
```

Comment:
This name isn't clear. What is `expression` as a duration, and why/how is it different from frequency?

Reply:
renamed `frequency`